code/controllers/subsystem/garbage.dm

```diff
@@ -168,7 +169,10 @@ var/datum/subsystem/garbage_collector/SSgarbage
 		del(D)
 	else if(isnull(D.gc_destroyed))
 		D.gc_destroyed = GC_CURRENTLY_BEING_QDELETED
+		var/start_time = world.time
 		var/hint = D.Destroy(force) // Let our friend know they're about to get fucked up.
+		if(world.time != start_time)
```

Comment:
http://www.byond.com/docs/ref/info.html#/world/var/time
`Also, if the server runs overtime during a tick (because procs take longer than tick_lag to finish) this still only counts as one tick.`